dashboard/app/models/user.rb

```diff
@@ -620,8 +620,16 @@ def user_level_locked?(script_level, level)
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
 
-    # Find the user level that we've most recently had progress on
-    user_level = user_levels_by_level.values.flatten.sort_by!(&:updated_at).last
+    # some of our user_levels may be for levels within level_groups, or for levels
+    # that are no longer in this script. we want to ignore those, and only look
+    # user_levels that have matching script_levels
+    levels = script.script_levels.map(&:level_ids).flatten
+    user_levels_by_level = user_levels.
+      where(script_id: script.id, level: levels).
+      index_by(&:level_id)
+
+    # Find the user_level that we've most recently had progress on
+    user_level = user_levels_by_level.values.max_by(&:updated_at)
```

Comment:
It's also worth noting that this method is likely failing silently in many cases in production right now. Most of the time, when we fail to find a `last_script_level`, we're okay because there is at least one unpassed progression level elsewhere in the script. In the case where that is not true, this method returns nil, and then we get HB errors.